src/Extension.php

```diff
@@ -72,29 +88,35 @@ public function configure(ArrayNodeDefinition $builder)
      */
     public function load(ContainerBuilder $container, array $config)
     {
-        $loader = new XmlFileLoader($container, new FileLocator(__DIR__.'/Resources/config'));
-        $loader->load('services.xml');
+        $loader = new PhpFileLoader($container, new FileLocator(__DIR__.'/Resources/config'));
+        $loader->load('services.php');
 
-        if (isset($config['migrations'])) {
-            $config['migrations'] = (array) $config['migrations'];
-
-            if (!class_exists('Doctrine\DBAL\Migrations\Migration')) {
-                throw new \RuntimeException('Configuration requires doctrine/migrations package');
-            }
+        $container->setParameter('behat.doctrine_data_fixtures.use_backup', $config['use_backup']);
+        if ($config['use_backup']) {
+            $loader->load('backup.php');
         }
 
-        $container->setParameter('behat.doctrine_data_fixtures.autoload', $config['autoload']);
-        $container->setParameter('behat.doctrine_data_fixtures.directories', $config['directories']);
-        $container->setParameter('behat.doctrine_data_fixtures.fixtures', $config['fixtures']);
-        $container->setParameter('behat.doctrine_data_fixtures.lifetime', $config['lifetime']);
-        $container->setParameter('behat.doctrine_data_fixtures.migrations', $config['migrations']);
-        $container->setParameter('behat.doctrine_data_fixtures.use_backup', $config['use_backup']);
+        $keys = ['autoload', 'directories', 'fixtures', 'lifetime'];
+        foreach ($keys as $key) {
+            $container->setParameter('behat.doctrine_data_fixtures.'.$key, $config[$key]);
+        }
     }
 
     /**
      * {@inheritdoc}
      */
     public function process(ContainerBuilder $container)
     {
+        //Backup Services
+        if ($container->hasDefinition(BackupService::class)) {
```

Comment:
I think early return is a bit more readable, like in official documentation

https://symfony.com/doc/3.4/service_container/tags.html#create-a-compiler-pass